hieradata/clients/azure.ci.jenkins.io.yaml

```diff
@@ -352,6 +352,22 @@ profile::jenkinscontroller::jcasc:
               - <powershell>
               - "(Get-Content C:\\ProgramData\\Datadog\\datadog.yaml -Raw) -Replace 'api_key:', 'api_key: ENC[PKCS7,MIIBmQYJKoZIhvcNAQcDoIIBijCCAYYCAQAxggEhMIIBHQIBADAFMAACAQEwDQYJKoZIhvcNAQEBBQAEggEAqyumVXNZAJylM6NiC/zKT+FKHqGnSwfUMWBupxumSJWL8GHEYFPB/Irr+sRiCPkRWW88bYg7Mp3Rv3qbxc0Aw/cU3XcQZB8RrAOsODBMjOJpHTvoB5aDtSYqygCjffPgwXK0DKguR2ADtsNkMcK7BOT02uaXgZOwo2S5UKkuDn6DsQBq9m1Ho8SwKw/E/+rwsUow+o5JVvPF9//8BIb53Y+wHYm2tOSe5dHFzBIP/+mJMHMpYiE5/q5LBELM9ZXhtBx2zeagEzbriDdlnDkMZk5AMWYz3+omDIX1MO2Fmg9Ak6O4dcNl4hhGJ5KtmPgLGzuhvfh018gi3NVhMBXv9zBcBgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBBltum5EYpT8EqN+vY0wudsgDBX++BA5bC+DpDxuqoANdKkUZdPyBo2cwxq8W4FfUtKE6ZReUIXxvARdR/Z2c5gYY4=]' | Set-Content C:\\ProgramData\\Datadog\\datadog.yaml"
               - </powershell>
+          - description: "Windows 2022"
+            maxInstances: 10
+            instanceType: T3Xlarge # 4 vCPUs / 16 Gb
+            os: "windows"
+            os_version: "2022"
+            architecture: "amd64"
+            labels:
+              - docker-windows-2022
+              - windows-2022
+            useAsMuchAsPossible: true
+            spot: true
```

Comment:
Not directly related to this PR, do we really want "spot" machine for Windows?
Windows builds are usually long, not sure if taking the risk of getting them interrupted at any time is the right choice.

Reply:
Good point. It won't kill to switch them, like we did for highmem VM agents, to "normal" workload (more expansive of course).

@lemeurherve can you open a PR for setting ALL windows templates to be non spot? Don't forget this PR :)